File: Python3/프로그래머스/1/178871. 달리기 경주/달리기 경주.py
def solution(players, callings):
    rank = {name: i for i, name in enumerate(players)}

    for name in callings:
        i = rank[name]         
        players[i], players[i-1] = players[i-1], players[i] 
        rank[players[i]] = i
        rank[players[i-1]] = i - 1

    return players

File: Java/프로그래머스/1/178871. 달리기 경주/달리기 경주.java
import java.util.*;

class Solution {
    public String[] solution(String[] players, String[] callings) {
        HashMap<String, Integer> rank = new HashMap<>();
        
        for (int i=0; i < players.length; i++) {
            rank.put( players[i] , i);
        }
        
        for (String name : callings){
            int k = rank.get(name);
            
            String frontPlayer = players[k-1];
            players[k]=frontPlayer;
            players[k-1]=name;
            
            rank.put(frontPlayer,k);
            rank.put(name,k-1);
        }
        
        return players;
    }
}